[tp] fix torch compile regression

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #111521

The most recent refactor of TP
https://github.com/pytorch/pytorch/pull/111160 breaks torch compile
path, so reverting the behavior back by:
1. use the old default prepare_input/output
2. add the colwise/rowwise parallel test instead

wanchaol committed:
[tp] fix torch compile regression

The most recent refactor of TP
https://github.com/pytorch/pytorch/pull/111160 breaks torch compile
path, so reverting the behavior back by:
1. use the old default prepare_input/output
2. add the colwise/rowwise parallel test instead

[ghstack-poisoned]

fduwjj commented on torch/distributed/tensor/parallel/style.py:
```diff
@@ -482,8 +482,8 @@ class RowwiseParallel(ParallelStyle):
 
     def __init__(
         self,
-        _prepare_input=None,
-        _prepare_output=None,
```
But this somehow revert the `input_layouts` and `output_layouts`? So does this mean when users want to use `input_layouts` and `output_layouts`, they have to set `_prepare_input` to `None`?

wanchaol replied:
This is just a quick fix, I'll do a big rewrite on top to make sure user behavior won't change

fduwjj replied:
I see.

wanchaol replied:
OK I gave up on the big rewrite and found a quicker solution right now, however it does not fundamentally fix the torch.compile problem in the new style `input_layouts` and `output_layouts` path (i.e. to configure sequence parallel torch.compile still fail). We should do a full rewrite soon to make the TP code cleaner and tracing friendly

wanchaol replied:
this "hack" preserves the intended behavior, but I dislike it..

wanchaol committed:
Update on "[tp] fix torch compile regression"


The most recent refactor of TP
https://github.com/pytorch/pytorch/pull/111160 breaks torch compile
path, so reverting the behavior back by:
1. use the old default prepare_input/output
2. add the colwise/rowwise parallel test instead

[ghstack-poisoned]

wanchaol commented:
@pytorchbot merge

wanchaol committed:
Update on "[tp] fix torch compile regression"


The most recent refactor of TP
https://github.com/pytorch/pytorch/pull/111160 breaks torch compile
path, so reverting the behavior back by:
1. use the old default prepare_input/output
2. add the colwise/rowwise parallel test instead

[ghstack-poisoned]

wanchaol commented:
@pytorchbot merge
